core/src/main/java/dev/streamx/cli/ingestion/publish/payload/PayloadResolver.java

```diff
@@ -3,120 +3,161 @@
 import dev.streamx.cli.ingestion.publish.payload.source.SourceResolver;
 import dev.streamx.cli.ingestion.publish.payload.typed.SourceType;
 import dev.streamx.cli.ingestion.publish.payload.typed.TypedPayloadFragmentResolver;
+import dev.streamx.cli.util.CollectionUtils;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 import java.io.IOException;
 import java.util.List;
+import java.util.Optional;
 import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
 
 @ApplicationScoped
 public class PayloadResolver {
 
-  @Inject
-  InitialPayloadResolver initialPayloadResolver;
-
   @Inject
   SourceResolver sourceResolver;
 
   @Inject
   TypedPayloadFragmentResolver typedPayloadFragmentResolver;
 
   @Inject
-  ValueReplacementExtractor valueReplacementExtractor;
+  JsonPathExtractor jsonPathExtractor;
+
+  @Inject
+  @PayloadProcessing
+  ObjectMapper objectMapper;
 
   public JsonNode createPayload(List<PayloadArgument> payloadArguments) {
-    try {
-      return doCreatePayload(payloadArguments);
-    } catch (IOException e) {
-      throw PayloadException.ioException(e);
+    if (CollectionUtils.isEmpty(payloadArguments)) {
+      throw PayloadException.payloadNotFound();
     }
-  }
-
-  private JsonNode doCreatePayload(List<PayloadArgument> payloadArguments)
-      throws IOException {
-    InitialPayload result = initialPayloadResolver.computeInitialPayload(payloadArguments);
 
-    String initialData = result.initialData();
-    DocumentContext documentContext = prepareInitialDocument(initialData);
+    DocumentContext documentContext = prepareInitialDocument();
 
-    replaceValues(documentContext, result.replacements());
+    documentContext = mergePayloads(documentContext, payloadArguments);
 
     return documentContext.json();
   }
 
-  private DocumentContext prepareInitialDocument(String initialData) {
-    try {
-      JsonNode initialJson = extractPayloadFragment(
-          PayloadArgument.ofJsonNode(initialData), initialData);
+  private DocumentContext prepareInitialDocument() {
+    JsonNode initialJson = objectMapper.createObjectNode();
 
-      return JsonPath.parse(initialJson);
-    } catch (JsonParseException exception) {
-      throw PayloadException.jsonParseException(exception, initialData);
-    } catch (JsonProcessingException exception) {
-      throw PayloadException.genericJsonProcessingException(exception, initialData);
-    } catch (IOException e) {
-      throw PayloadException.ioException(e);
-    }
+    return JsonPath.parse(initialJson);
   }
 
-  private void replaceValues(DocumentContext documentContext,
+  private DocumentContext mergePayloads(DocumentContext documentContext,
       List<PayloadArgument> payloadArguments) {
-    if (payloadArguments == null || payloadArguments.isEmpty()) {
-      return;
-    }
 
     for (PayloadArgument payloadArgument : payloadArguments) {
       String value = payloadArgument.getValue();
```

Comment:
Please find better name for this variable (in worst case please use payloadArgumentValue) and its all use places in method parameters.